tests/virt_autotest/virsh_internal_snapshot.pm

```diff
@@ -33,6 +33,12 @@ sub run_test {
     return unless check_var("REGRESSION", "qemu-hypervisor") || check_var("SYSTEM_ROLE", "kvm");
 
     foreach my $guest (keys %xen::guests) {
+        my $guest_disk_type = script_output("virsh dumpxml $guest | grep \"<driver \" | grep -o \"type='.*'\" | cut -d \"'\" -f2");
+        if ($guest_disk_type =~ /raw/) {
+            record_info "INFO", "SKIP Snapshot test if guest disk type as raw";
+            return;    #RAW do not support Snapshot, so skip Snapshot test if guest disk type as RAW
+        }
+
```

Comment:
The two files are repeating the same code.Can you wrap it to a function?

Reply:
Great, got it.
I will update code to wrap it to a function.

Reply:
had updated code already